packages/@vue/cli-plugin-pwa/lib/HtmlPwaPlugin.js

```diff
@@ -185,8 +187,12 @@ function makeTag (tagName, attributes, closeTag = false) {
 
 function getTagHref (publicPath, href, assetsVersionStr) {
   let tagHref = `${href}${assetsVersionStr}`
-  if (!(/(http(s?)):\/\//gi.test(href))) {
+  if (!isHrefAbsoluteUrl(href)) {
     tagHref = `${publicPath}${tagHref}`
   }
   return tagHref
 }
+
+function isHrefAbsoluteUrl (href) {
+  return /(http(s?)):\/\//gi.test(href)
```

Comment:
Instead of a function like this, why not use a package that does url validation?

Reply:
I don't think it's a good practice to introduce extra dependencies for one-liners

Reply:
I am more worried about this part not being correct and exhaustive. What about if the path is given as `www.example.com`? Isn't it better to have a package which correctly understands the url and tells us about it?

Reply:
`www.example.com` is a relative path in the context of HTML, and it will fail this test as expected.

Other general-purpose packages may not serve the same interest as the pwa plugin. For example, the widely-used [`is-absolute-url`](https://github.com/sindresorhus/is-absolute-url) package (used by webpack-dev-server, postcss-normalize-url, etc.) But it treats `mainto:` and `file:` links as absolute URLs, too. That's correct according to the URL RFC, but not the case for PWA usage.

So it's better to contain such utility function in our own package, and write our own tests if really needed.

Reply:
You are right. I stand corrected.